src/amcrest/special.py

```diff
@@ -153,3 +155,62 @@ def mjpg_stream(self, channelno=None, typeno=None, path_file=None):
                 raise CommError(error)
 
         return ret.raw
+
+    def get_event_stream(self, eventcode='VideoMotion', callback=None,
+                         timeout=(TIMEOUT_HTTP_PROTOCOL, 3600*24*365)):
+        """
+        Params:
+        eventcode:
+        string list of events
+        eg 'VideoMotion,VideoLoss'
+        NOTE: No spaces allowed!
+
+        VideoMotion: motion detection event
+        VideoLoss: video loss detection event
+        VideoBlind: video blind detection event
+        AlarmLocal: alarm detection event
+        StorageNotExist: storage not exist event
+        StorageFailure: storage failure event
+        StorageLowSpace: storage low space event
+        AlarmOutput: alarm output event
+
+        callback:
+        function to use to process data when running in a Thread
+        event data is passed as the first argument
+
+        timeout:
+        int or tuple, requests timeout format
+        no-data timeout, set to a long value for streaming
+        """
+        try:
+            ret = self.command(
+                'eventManager.cgi?action=attach&codes=[{0}]'.format(eventcode),
+                stream=True,
+                timeout_cmd=timeout
+            )
+            if ret.encoding is None:
+                ret.encoding = 'utf-8'  #required by requests when streaming
+            while True:
+                for line in ret.iter_lines(3, decode_unicode=True):
+                    if line.lower().startswith('content-length'):
+                        cont_len = int(''.join(x for x in line
+                                               if x.isdigit()))
+                        # pause the generator
+                        break
+
+                # Continue the generator and read the exact amount of the body
+                # (plus 2 as there are 2 returns included).
+                for text in ret.iter_content(cont_len+2, decode_unicode=True):
+                    if callback:
+                        callback(text.strip())
+                    else:
+                        return text.strip()
+                    break
```

Comment:
I found that because each line is terminated with `'\r\n'` the existing `iter_lines` method doesn't work too well, no matter what is used for `chunk_size`. (I.e., it can always split right in the middle, and see two line delimiters when there really is only one.) So I came up with this generator instead:
```python
def lines(ret):
    line = ''
    for char in ret.iter_content(decode_unicode=True):
        line = line + char
        if line.endswith('\r\n'):
            yield line.strip()
            line = ''
```
Now with that generator the code can be simplified to:
```python
for line in lines(ret):
    if line.lower().startswith('content-length:'):
        text = next(ret.iter_content(chunk_size=int(line.split(':')[1]), decode_unicode=True))
        if callback:
            callback(text)
        else:
            return text
```
The `'\r\n\r\n'` that follows the eventInfo will just result in two blank lines returned by `lines()`, which shouldn't be an issue.